src/greenlight/step.clj

```diff
@@ -95,12 +95,14 @@
   (create-foo
     {:foo/name \"Foo 1\"}
     :output :foo.1/id)"
-  [step-name docstring & default-config]
-  (when-not (string? docstring)
-    (throw (IllegalArgumentException.
-             "defstep expects a docstring for the step constructor")))
-  (let [default-config (qualify-config-keys
-                         (apply hash-map default-config))]
+  [step-name & step-decl]
+  (let [docstring (if (string? (first step-decl)) (first step-decl) "")
+        default-config (-> (if (string? (first step-decl))
+                             (rest step-decl)
+                             step-decl)
+                           (->> (apply hash-map))
+                           (update :title #(or % (str *ns* \/ step-name)))
```

Comment:
This may drop an existing `::step/title` if provided, as we will have both `:title` and `::step/title` before `qualify-config-keys`.

Including the namespace seems a bit heavy, what about just defaulting to the `step-name` as the title if not provided?

Reply:
idk about using only the `step-name`. In my experience, the namespace usually also carries meaning, specially when a step is used in more than one namespace. 

Also, thanks for the other comments!